2020/10/27/craft-a-cms-for-developers.html.md

```diff
@@ -0,0 +1,234 @@
+
+3. Once in there, run `composer create-project craftcms/craft ./install` and that will use Composer to install Craft. In other words, it will download all of the files that Craft needs to run. You should see something like this at the end:
+
+```
+> @php craft setup/welcome
+
+   ______ .______          ___       _______ .___________.
+  /      ||   _  \        /   \     |   ____||           |
+ |  ,----'|  |_)  |      /  ^  \    |  |__   `---|  |----`
+ |  |     |      /      /  /_\  \   |   __|      |  |
+ |  `----.|  |\  \----./  _____  \  |  |         |  |
+  \______|| _| `._____/__/     \__\ |__|         |__|
+ 
+     A       N   E   W       I   N   S   T   A   L   L
+               ______ .___  ___.      _______.
+              /      ||   \/   |     /       |
+             |  ,----'|  \  /  |    |   (----`
+             |  |     |  |\/|  |     \   \
+             |  `----.|  |  |  | .----)   |
+              \______||__|  |__| |_______/
+
+
+Generating an application ID ... done (CraftCMS--xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx)
+Generating a security key ... done (xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx)
+
+Welcome to Craft CMS! Run the following command if you want to setup Craft from your terminal:
+
+    /var/www/install/craft setup
+```
+
+4. After that's done, use this bit of Linux black magic `(shopt -s dotglob; mv -v ./install/* .)`. This command moves all the files that composer just downloaded into `/var/www/install` out from there and into `/var/www`. Then `rm -r install` to remove the `install` directory because we will no longer need it.
+
+5. Now do `php craft setup` to use Craft's CLI to set up the site, its configuration and database structure. Just follow the prompts. When it asks you for database configuration, choose `mysql` as the database driver, set `mysql` as the database server name (because that's the name we've given it in our `docker-compose.yml`), and use the environment variables defined in `docker-compose.yml` around lines 14 to 17 for the rest of the values. You can change these to whatever you want as long as you make sure that it coincides with how you defined your MySQL database container in the `docker-compose.yml` file. The prompts should look something like this:
+
+```
+Which database driver are you using? [mysql,pgsql,?]: mysql
+Database server name or IP address: [127.0.0.1] mysql 
+Database port: [3306] 
+Database username: [root] craft
+Database password: 
+Database name: craft_demo
+Database table prefix: 
+Testing database credentials ... success!
+Saving database credentials to your .env file ... done
+```
+
+6. Next, sort out some Craft file permission requirements with these: `chmod -R o+w config && chmod -R o+w storage && chmod -R o+w web/cpresources`. These are just some directories that Craft needs write access to.
```

Comment:
`chmod` takes a list of files/directories to modify, so these three `chmod` commands can be combined:

```
chmod -R o+w config storage web/cpresources
```

Reply:
Oh I see. Changed it. Thanks!